fitlins/interfaces/utils.py

```diff
@@ -17,16 +20,20 @@ def __init__(self, fields=None):
     def _add_output_traits(self, base):
         return add_traits(base, self._fields)
 
+    def _calculate_length(self, val):
+        _lengths = list(map(len, val))
+        if self._lengths is None:
+            self._lengths = _lengths
+        elif _lengths != self._lengths:
+            raise ValueError("List lengths must be consistent across fields")
+        self._lengths = _lengths
+
     def _list_outputs(self):
         outputs = self._outputs().get()
-        lengths = None
         for key in self._fields:
             val = getattr(self.inputs, key)
-            _lengths = list(map(len, val))
-            if lengths is None:
-                lengths = _lengths
-            elif _lengths != lengths:
-                raise ValueError("List lengths must be consistent across fields")
+            if self._check_lengths is True:
+                self._calculate_length(val)
             outputs[key] = [elem for sublist in val for elem in sublist]
```

Comment:
This may be excessive caution, but just thinking about re-running the same object with changed inputs, we might want to reset this value after using it.

```suggestion
            outputs[key] = [elem for sublist in val for elem in sublist]
        self._lengths = None
```

Reply:
Ok sounds good.